### ISSUE TITLE
Gitea 1.16.8 / git-as-svn 2.3.0 : io.gitea.ApiException: Forbidden

### ISSUE THREAD

**Issue by simchanu29** [git-as-svn/git-as-svn#530]
I tried to setup git-as-svn v2.3.0 for an existing gitea server (v1.16.8).
Gitea and git-as-svn are running on the same machine and git-as-svn has access with the user git to these repositories.

My configuration file (I replaced sensitive informations with <my token> and <my host>)
```
!config

cacheConfig: !persistentCache
  path: /var/cache/git-as-svn/git-as-svn.mapdb

repositoryMapping: !giteaMapping
  path: /home/git/gitea-repositories

shared:
  - !gitea
    url: https://<my host>/api/v1
    lfs: false
    token: <my token>

```

- I generated the token from the Application tab from the configuration panel of a user. So it should be ok.
- I enabled debug mode on log4j.

git-as-svn.error.log is empty

I have a bare git repository created on Gitea by my user svn. I can checkout, ls but can't commit to the repository.

git-as-svn.log for a commit (I changed the ip addresses to x.y.z.w) : 
```
2022-07-01T13:41:46,527  INFO [SvnServer-thread-5] (SvnServer.kt:147) - Connection from /x.y.z.w:45492 closed
2022-07-01T13:47:05,211  INFO [SvnServer-thread-6] (SvnServer.kt:84) - New connection from: /x.y.z.w:45494
2022-07-01T13:47:05,221  INFO [SvnServer-thread-6] (SvnServer.kt:191) - Client: SVN/1.13.0 (x86_64-pc-linux-gnu)
2022-07-01T13:47:05,633  INFO [SvnServer-thread-6] (SvnServer.kt:244) - User: $anonymous
2022-07-01T13:47:05,633 DEBUG [SvnServer-thread-6] (FileSnapshot.java:497) - file=/home/git/.config/jgit/config, isRacyClean=false, read=2022-07-01 13:35:05.060593493, lastModified=2022-06-29 11:51:32.148942763, delta=179012911650730 ns, racy<=2500000000 ns
2022-07-01T13:47:05,633 DEBUG [SvnServer-thread-6] (FileSnapshot.java:541) - file=/home/git/.config/jgit/config, is unmodified
2022-07-01T13:47:05,652 DEBUG [SvnServer-thread-6] (FileSnapshot.java:497) - file=/home/git/.gitconfig, isRacyClean=false, read=2022-07-01 13:35:05.184704468, lastModified=2021-05-27 10:06:56.356615248, delta=34572488828089220 ns, racy<=2500000000 ns
2022-07-01T13:47:05,652 DEBUG [SvnServer-thread-6] (FileSnapshot.java:541) - file=/home/git/.gitconfig, is unmodified
2022-07-01T13:47:05,653 DEBUG [SvnServer-thread-6] (FileSnapshot.java:497) - file=/home/git/gitea-repositories/svn/test.git/config, isRacyClean=false, read=2022-07-01 13:40:18.566333629, lastModified=2022-07-01 13:40:15.956662462, delta=2609671167 ns, racy<=2500000000 ns
2022-07-01T13:47:05,653 DEBUG [SvnServer-thread-6] (FileSnapshot.java:541) - file=/home/git/gitea-repositories/svn/test.git/config, is unmodified
2022-07-01T13:47:05,672 DEBUG [SvnServer-thread-6] (FileSnapshot.java:534) - file=/home/git/gitea-repositories/svn/test.git/refs/git-as-svn/v1/master.lock, cannot be racily clean
2022-07-01T13:47:05,691 DEBUG [SvnServer-thread-6] (FileSnapshot.java:497) - file=/home/git/.config/jgit/config, isRacyClean=false, read=2022-07-01 13:35:05.060593493, lastModified=2022-06-29 11:51:32.148942763, delta=179012911650730 ns, racy<=2500000000 ns
2022-07-01T13:47:05,692 DEBUG [SvnServer-thread-6] (FileSnapshot.java:541) - file=/home/git/.config/jgit/config, is unmodified
2022-07-01T13:47:05,692 DEBUG [SvnServer-thread-6] (FileSnapshot.java:497) - file=/home/git/.gitconfig, isRacyClean=false, read=2022-07-01 13:35:05.184704468, lastModified=2021-05-27 10:06:56.356615248, delta=34572488828089220 ns, racy<=2500000000 ns
2022-07-01T13:47:05,692 DEBUG [SvnServer-thread-6] (FileSnapshot.java:541) - file=/home/git/.gitconfig, is unmodified
2022-07-01T13:47:05,693 DEBUG [SvnServer-thread-6] (FileSnapshot.java:497) - file=/home/git/gitea-repositories/svn/test.git/config, isRacyClean=false, read=2022-07-01 13:40:18.566333629, lastModified=2022-07-01 13:40:15.956662462, delta=2609671167 ns, racy<=2500000000 ns
2022-07-01T13:47:05,711 DEBUG [SvnServer-thread-6] (FileSnapshot.java:541) - file=/home/git/gitea-repositories/svn/test.git/config, is unmodified
2022-07-01T13:47:05,721 DEBUG [SvnServer-thread-6] (SvnServer.kt:129) - Receive command: commit
2022-07-01T13:47:11,943  INFO [SvnServer-thread-6] (SvnServer.kt:244) - User: svn
2022-07-01T13:47:12,029  INFO [SvnServer-thread-6] (SvnServer.kt:147) - Connection from /x.y.z.w:45494 closed
```

I get these logs from `journalctl -u git-as-svn` : 
```
Jul 01 13:47:12 Karto1 git-as-svn[145959]: Exception in thread "SvnServer-thread-6" java.lang.IllegalStateException: java.util.concurrent.ExecutionException: io.gitea.ApiException: Forbidden
Jul 01 13:47:12 Karto1 git-as-svn[145959]:         at svnserver.ext.gitea.mapping.GiteaAccess.getCachedProject(GiteaAccess.kt:83)
Jul 01 13:47:12 Karto1 git-as-svn[145959]:         at svnserver.ext.gitea.mapping.GiteaAccess.canWrite(GiteaAccess.kt:50)
Jul 01 13:47:12 Karto1 git-as-svn[145959]:         at svnserver.repository.VcsAccess$DefaultImpls.checkWrite(VcsAccess.kt:35)
Jul 01 13:47:12 Karto1 git-as-svn[145959]:         at svnserver.ext.gitea.mapping.GiteaAccess.checkWrite(GiteaAccess.kt:32)
Jul 01 13:47:12 Karto1 git-as-svn[145959]:         at svnserver.server.SessionContext.checkWrite(SessionContext.kt:138)
Jul 01 13:47:12 Karto1 git-as-svn[145959]:         at svnserver.server.command.CommitCmd.permissionCheck(CommitCmd.kt:75)
Jul 01 13:47:12 Karto1 git-as-svn[145959]:         at svnserver.server.command.CommitCmd.permissionCheck(CommitCmd.kt:54)
Jul 01 13:47:12 Karto1 git-as-svn[145959]:         at svnserver.server.command.BaseCmd.process$lambda-1(BaseCmd.kt:36)
Jul 01 13:47:12 Karto1 git-as-svn[145959]:         at svnserver.server.step.CheckPermissionStep.process(CheckPermissionStep.kt:40)
Jul 01 13:47:12 Karto1 git-as-svn[145959]:         at svnserver.server.SvnServer.serveClient(SvnServer.kt:122)
Jul 01 13:47:12 Karto1 git-as-svn[145959]:         at svnserver.server.SvnServer.run$lambda-2(SvnServer.kt:85)
Jul 01 13:47:12 Karto1 git-as-svn[145959]:         at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
Jul 01 13:47:12 Karto1 git-as-svn[145959]:         at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
Jul 01 13:47:12 Karto1 git-as-svn[145959]:         at java.base/java.lang.Thread.run(Thread.java:833)
Jul 01 13:47:12 Karto1 git-as-svn[145959]: Caused by: java.util.concurrent.ExecutionException: io.gitea.ApiException: Forbidden
Jul 01 13:47:12 Karto1 git-as-svn[145959]:         at com.google.common.util.concurrent.AbstractFuture.getDoneValue(AbstractFuture.java:588)
Jul 01 13:47:12 Karto1 git-as-svn[145959]:         at com.google.common.util.concurrent.AbstractFuture.get(AbstractFuture.java:547)
Jul 01 13:47:12 Karto1 git-as-svn[145959]:         at com.google.common.util.concurrent.AbstractFuture$TrustedFuture.get(AbstractFuture.java:113)
Jul 01 13:47:12 Karto1 git-as-svn[145959]:         at com.google.common.util.concurrent.Uninterruptibles.getUninterruptibly(Uninterruptibles.java:204)
Jul 01 13:47:12 Karto1 git-as-svn[145959]:         at com.google.common.cache.LocalCache$Segment.getAndRecordStats(LocalCache.java:2235)
Jul 01 13:47:12 Karto1 git-as-svn[145959]:         at com.google.common.cache.LocalCache$Segment.loadSync(LocalCache.java:2201)
Jul 01 13:47:12 Karto1 git-as-svn[145959]:         at com.google.common.cache.LocalCache$Segment.lockedGetOrLoad(LocalCache.java:2159)
Jul 01 13:47:12 Karto1 git-as-svn[145959]:         at com.google.common.cache.LocalCache$Segment.get(LocalCache.java:2049)
Jul 01 13:47:12 Karto1 git-as-svn[145959]:         at com.google.common.cache.LocalCache.get(LocalCache.java:3855)
Jul 01 13:47:12 Karto1 git-as-svn[145959]:         at com.google.common.cache.LocalCache.getOrLoad(LocalCache.java:3878)
Jul 01 13:47:12 Karto1 git-as-svn[145959]:         at com.google.common.cache.LocalCache$LocalLoadingCache.get(LocalCache.java:4780)
Jul 01 13:47:12 Karto1 git-as-svn[145959]:         at svnserver.ext.gitea.mapping.GiteaAccess.getCachedProject(GiteaAccess.kt:78)
Jul 01 13:47:12 Karto1 git-as-svn[145959]:         ... 13 more
Jul 01 13:47:12 Karto1 git-as-svn[145959]: Caused by: io.gitea.ApiException: Forbidden
Jul 01 13:47:12 Karto1 git-as-svn[145959]:         at io.gitea.ApiClient.handleResponse(ApiClient.java:934)
Jul 01 13:47:12 Karto1 git-as-svn[145959]:         at io.gitea.ApiClient.execute(ApiClient.java:850)
Jul 01 13:47:12 Karto1 git-as-svn[145959]:         at io.gitea.api.RepositoryApi.repoGetByIDWithHttpInfo(RepositoryApi.java:8914)
Jul 01 13:47:12 Karto1 git-as-svn[145959]:         at io.gitea.api.RepositoryApi.repoGetByID(RepositoryApi.java:8900)
Jul 01 13:47:12 Karto1 git-as-svn[145959]:         at svnserver.ext.gitea.mapping.GiteaAccess$1.load(GiteaAccess.kt:118)
Jul 01 13:47:12 Karto1 git-as-svn[145959]:         at svnserver.ext.gitea.mapping.GiteaAccess$1.load(GiteaAccess.kt:91)
Jul 01 13:47:12 Karto1 git-as-svn[145959]:         at com.google.common.cache.LocalCache$LoadingValueReference.loadFuture(LocalCache.java:3451)
Jul 01 13:47:12 Karto1 git-as-svn[145959]:         at com.google.common.cache.LocalCache$Segment.loadSync(LocalCache.java:2200)
Jul 01 13:47:12 Karto1 git-as-svn[145959]:         ... 19 more
```

It seems like I have given the wrong token to git-as-svn.conf, but I've done it several times now with different tokens and I got the same result.  Could it be a bug on this side or am I doing something wrong ?

**Comment by NGenesis** [comment]
This may be fixed with #551 as the Gitea environment variable `GITEA_REPO_USER_NAME` was being incorrectly set as `GITEA_REPO_USER` causing commits to fail.

**Comment by slonopotamus** [comment]
Presumably fixed by #551. Unfortunately, I don't currently have enough time to manually test this. And we don't have automated tests for commit process :(
